### PR TITLE
Add KMA aligner

### PR DESCRIPTION
**Pull request by dfornika**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/kma/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+categories:
+- Next Gen Mappers
+description: Map with KMA
+homepage_url: https://bitbucket.org/genomicepidemiology/kma
+long_description: |
+  KMA is mapping a method designed to map raw reads directly against redundant databases, in an
```
Comment (bgruening) [inline tools/kma/.shed.yml]: remove "mapping"?

File: tools/kma/macros.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <xml name="kma_index">
+        <param label="Select a KMA index" name="kma_index" type="select">
+            <options from_data_table="kma_index">
+                <validator message="No KMA index available" type="no_options" />
+            </options>
+        </param>
+    </xml>
+    <xml name="citations">
+        <citations>
+            <citation type="bibtex">
```
Comment (bgruening) [inline tools/kma/macros.xml]: @dfornika can we add the DOI here instead? DOIs are more portable imho and we can do more fancy things with it.

File: tools/kma/kma_map.xml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            </when>
+            <when value="yes">
+                <param format="@INTYPES@" name="forward_input" type="data" label="Forward strand" help="FASTQ dataset"/>
+                <param format="@INTYPES@" name="reverse_input" type="data" label="Reverse strand" help="FASTQ dataset"/>
+            </when>
+            <when value="no">
+                <param format="@INTYPES@" label="Input sequences" name="input_sequences" type="data" help="FASTQ datasets"/>
+            </when>
+        </conditional>
+	<param name="kma_index" type="select">
```
Comment (bgruening) [inline tools/kma/kma_map.xml]: something is wrong here with indentation ...

File: tools/kma/kma_map.xml [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        #if str($settings.advanced) == "advanced" and $settings.matrix
+            && gunzip output.mat.gz
+        #end if
+        && gunzip output.frag.gz
+	]]>
+    </command>
+    <inputs>
+        <conditional name="single_paired">
+            <param name="single_paired_selector" type="select" label="Single or paired reads" help="--paired">
+                <option value="collection">Collection</option>
```
Comment (bgruening) [inline tools/kma/kma_map.xml]: Can we name this paired collection?

File: tools/kma/kma_map.xml [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+                <option value="yes">Paired</option>
```
Comment (bgruening) [inline tools/kma/kma_map.xml]: Paired end data

File: tools/kma/kma_map.xml [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+                <option selected="True" value="no">Single</option>
```
Comment (bgruening) [inline tools/kma/kma_map.xml]: Can we also use the value `single`

File: tools/kma/kma_map.xml [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+            <when value="simple">
+            </when>
+            <when value="advanced">
+                <param name="kmer_size" type="integer" min="4" value="16" max="32" label="Kmer Size" />
+                <param name="p_value" type="float" min="0.0" value="0.05" max="1.0" label="p-value"/>
+                <param name="exhaustive_mode" type="boolean" truevalue="-ex_mode" falsevalue="" label="Exhaustive Mode" />
+                <param name="decontaminate" type="boolean" truevalue="-deCon" falsevalue="" label="Decontaminate" />
+                <param name="dense" type="boolean" truevalue="-dense" falsevalue="" label="Do not allow insertions in assembly" />
+                <param name="ref_fsa" type="boolean" truevalue="-ref_fsa" falsevalue="" label="Consensus sequence has 'n' instead of gaps" />
+                <param name="matrix" type="boolean" truevalue="-matrix" falsevalue="" label="Print assembly matrix" />
```
Comment (bgruening) [inline tools/kma/kma_map.xml]: its not really print in our case, maybe output assembly matrix?

File: tools/kma/kma_map.xml [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+            </when>
+        </conditional>
+	<param name="kma_index" type="select">
+	    <options from_data_table="kma_index">
+	        <validator type="no_options" message="No KMA index available" />
+            </options>
+        </param>
+        <conditional name="settings">
+            <param name="advanced" type="select" label="Specify advanced parameters">
+                <option value="simple" selected="true">No, use program defaults.</option>
```
Comment (bgruening) [inline tools/kma/kma_map.xml]: I would remove the dot in the end.

File: tools/kma/kma_index.loc.sample [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# Expect three columns, tab separated, as follows:
```
Comment (bgruening) [inline tools/kma/kma_index.loc.sample]: can we add a comment here, how to create thos indeces ...
Than admins have it easier to create it manually if there is no data manager

Comment (bgruening) [reply tools/kma/kma_index.loc.sample]: Oh a DM is also there! You rock!

File: tools/kma/kma_map.xml (hunk 6 above)
Comment (dfornika) [reply tools/kma/kma_map.xml]: I agree those values are unclear, I copied from another tool (lazy me). I've changed to match your suggestions.

**Comment by dfornika** [comment]
@bgruening I think I've addressed everything(?)

File: tools/kma/kma_map.xml [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+        #end if
+        && gunzip output.frag.gz
+	]]>
+    </command>
+    <inputs>
+        <conditional name="single_paired">
+            <param name="single_paired_selector" type="select" label="Single or paired reads" help="--paired">
+                <option value="paired_collection">Paired collection</option>
+                <option value="paired">Paired-end data</option>
+                <option selected="True" value="single">Single</option>
```
Comment (bgruening) [inline tools/kma/kma_map.xml]: Single end data?

**Pull request merged**